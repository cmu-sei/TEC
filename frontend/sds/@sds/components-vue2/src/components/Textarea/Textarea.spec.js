// TEC - Automated Mismatch Detection Tool
// Copyright 2023 Carnegie Mellon University.
// NO WARRANTY. THIS CARNEGIE MELLON UNIVERSITY AND SOFTWARE ENGINEERING INSTITUTE MATERIAL IS FURNISHED ON AN "AS-IS" BASIS. CARNEGIE MELLON UNIVERSITY MAKES NO WARRANTIES OF ANY KIND, EITHER EXPRESSED OR IMPLIED, AS TO ANY MATTER INCLUDING, BUT NOT LIMITED TO, WARRANTY OF FITNESS FOR PURPOSE OR MERCHANTABILITY, EXCLUSIVITY, OR RESULTS OBTAINED FROM USE OF THE MATERIAL. CARNEGIE MELLON UNIVERSITY DOES NOT MAKE ANY WARRANTY OF ANY KIND WITH RESPECT TO FREEDOM FROM PATENT, TRADEMARK, OR COPYRIGHT INFRINGEMENT.
// Released under a BSD (SEI)-style license, please see license.txt or contact permission@sei.cmu.edu for full terms.
// [DISTRIBUTION STATEMENT A] This material has been approved for public release and unlimited distribution.  Please see Copyright notice for non-US Government use and distribution.
// This Software was created using the SEI Design System and includes and/or makes use of Third-Party Software subject to its own license.
// DM23-003
 
 
import { shallowMount } from '@vue/test-utils'
import Component from "./Textarea.vue";

describe("Textarea.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });

  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with maxlength and count-characters prop assigned", () => {
    const propsData = {
      maxlength: 50,
      countCharacters: true
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with rows prop assigned", () => {
    const propsData = {
      rows: 2,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with disabled prop assigned", () => {
    const propsData = {
      disabled: true,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with readonly prop assigned", () => {
    const propsData = {
      readonly: true,
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with placeholder prop assigned", () => {
    const propsData = {
      placeholder: "This is a placeholder",
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("matches snapshot with value prop assigned", () => {
    const propsData = {
      value: "This is the text content",
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
