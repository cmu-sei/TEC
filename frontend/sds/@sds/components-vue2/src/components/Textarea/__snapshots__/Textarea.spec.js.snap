// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea.vue matches snapshot with disabled prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"5\\" maxlength=\\"524288\\" disabled=\\"disabled\\" class=\\"form-control\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue matches snapshot with maxlength and count-characters prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"5\\" maxlength=\\"50\\" class=\\"form-control\\"></textarea>
  <character-counter-stub currentvalue=\\"0\\" maxvalue=\\"50\\" class=\\"text-right text-gray-500\\"></character-counter-stub>
</div>"
`;

exports[`Textarea.vue matches snapshot with no props assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"5\\" maxlength=\\"524288\\" class=\\"form-control\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue matches snapshot with placeholder prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"5\\" maxlength=\\"524288\\" placeholder=\\"This is a placeholder\\" class=\\"form-control\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue matches snapshot with readonly prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"5\\" maxlength=\\"524288\\" readonly=\\"readonly\\" class=\\"form-control\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue matches snapshot with rows prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"2\\" maxlength=\\"524288\\" class=\\"form-control\\"></textarea>
  <!---->
</div>"
`;

exports[`Textarea.vue matches snapshot with value prop assigned 1`] = `
"<div data-id=\\"sds-textarea\\"><textarea rows=\\"5\\" maxlength=\\"524288\\" class=\\"form-control\\"></textarea>
  <!---->
</div>"
`;
